### PR TITLE
fix: Do compressBinlog to fix logID 0

### PR DESCRIPTION
**Pull request by bigsheeper**
issue: https://github.com/milvus-io/milvus/issues/34059

Do compressBinlog to ensure that reloadFromKV will fill binlogs' logID after datacoord restarts.

### CODE REVIEW

File: internal/metastore/kv/datacoord/kv_catalog.go [hunk 1]
```diff
@@ -217,29 +217,26 @@ func (kc *Catalog) applyBinlogInfo(segments []*datapb.SegmentInfo, insertLogs, d
...
 
 		if len(segmentInfo.Deltalogs) == 0 {
 			segmentInfo.Deltalogs = deltaLogs[segmentInfo.ID]
-		} else {
-			err = binlog.CompressFieldBinlogs(segmentInfo.Deltalogs)
-			if err != nil {
-				return err
-			}
+		}
+		err = binlog.CompressFieldBinlogs(segmentInfo.Deltalogs)
```
Comment (chyezh) [inline internal/metastore/kv/datacoord/kv_catalog.go:229]: if err := ...; err != nil {return err}

Comment (bigsheeper) [reply internal/metastore/kv/datacoord/kv_catalog.go]: updated

**Pull request merged**